R/teal_slices.R

```diff
@@ -267,13 +215,18 @@ format.teal_slices <- function(x, show_all = FALSE, trim_lines = TRUE, ...) {
   checkmate::assert_flag(show_all)
   checkmate::assert_flag(trim_lines)
 
-  slices_list <- slices_to_list(x)
+  x <- as.list(x, recursive = TRUE)
+  attrs <- attributes(x)
+  attributes(x) <- NULL
+  slices_list <- list(slices = x, attributes = attrs)
+  slices_list <- Filter(Negate(is.null), slices_list) # drop attributes if empty
```

Comment:
I am having seconds thoughts if we should keep empty attributes (as it happens now) and I reckon they can be removed

Reply:
```suggestion

```
but this will never empty as `teal_slices` has default non-NULL-able parameter `allow_add = TRUE` hence this always fill be in attributes

Reply:
I thought about it. While this is true now, perhaps we will allow having no attributes in the future. Some attributes are only listed if they are not defaults, e.g. `count_type`.

Reply:
Yeah, but in the future if you remove `all_add` parameter from `teal_slices` it will have an extra implication of attributes removal from `print`, `format` and `store`. No other argument will have this implication. And you need to remember about it. I think it's gonna be less painful in the future to bring back this `Filter()` if we allow all attributes to be NULL, then figure out why default call for `print` for `teal_slices` dropped attributes. This is minor and I defer to you to make a final decision

Reply:
Hold on. This is not about not printing NULL attributes, NULL attributes are never printed explicitly. This `Filter` is only here because we explicitly extract attributes and that can **add** a NULL to the list that is passed to `do.call`.

Reply:
```
slices <- as.list(letters[1:3])
attrs <- NULL
list(slices = slices, attrs = attrs)
```
```
$slices
$slices[[1]]
[1] "a"

$slices[[2]]
[1] "b"

$slices[[3]]
[1] "c"


$attrs
NULL
```

Better safe than sorry :slightly_smiling_face: